### PR TITLE
feat: Landscape Profile: Affiliation Step

### LINKED ISSUE #569: Landscape Profile: Step 4 - Affiliation
## Description
- Add partneship fields and radio buttons
- Add affiliation groups component
- Add organizations list component
- Add Partnership and Affiliation card to Profile
- Add Partnership and Affiliation to Landascape About card

## Additional context

Figma: https://www.figma.com/file/Uv20vRx7hDmeDw4c5dOhiY/Create%2FUpdate-Landscape-Profile_V2-R1.1?node-id=0%3A1

### PR DESCRIPTION
**Pull request by josebui**
### Checklist
- [x] Corresponding issue has been opened

### Related Issues
Fixes #569 

### Verification steps

Screen Shot 2022-10-25 at 1 50 54 PM
Screen Shot 2022-10-25 at 1 50 22 PM
Screen Shot 2022-10-25 at 1 50 10 PM

### CODE REVIEW

File: src/localization/locales/en-US.json [hunk 1]
```diff
@@ -355,7 +356,26 @@
...
-        "form_profile_commondities_info": "Which agricultural products and commodities are produced in your landscapes?"
+        "form_profile_commondities_info": "Which agricultural products and commodities are produced in your landscapes?",
+        "profile_affiliation_card_title": "Partnerships & Affiliations",
+        "profile_affiliation_card_partnership_description": "Landscape partnership since {{year}}",
+        "profile_affiliation_card_affiliated_groups_title": "Affiliated Organizations",
+        "profile_affiliation_card_organizations_title": "Active Organizations",
+        "profile_affiliation_card_update_button": "Update Parntership Info",
+        "form_profile_partnership_status": "Is there a landscape partnership managing this landscape?",
+        "profile_profile_card_partnership_status_yes": "Yes",
+        "profile_profile_card_partnership_status_in_progress": "In-progress",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:367]:
```suggestion
        "profile_profile_card_partnership_status_in_progress": "In progress",
```

File: src/localization/locales/en-US.json [hunk 2]
```diff
@@ -355,7 +356,26 @@
...
         "form_profile_commondities": "Commodities",
         "form_profile_commondities_placeholder": "Select one or more products",
-        "form_profile_commondities_info": "Which agricultural products and commodities are produced in your landscapes?"
+        "form_profile_commondities_info": "Which agricultural products and commodities are produced in your landscapes?",
+        "profile_affiliation_card_title": "Partnerships & Affiliations",
+        "profile_affiliation_card_partnership_description": "Landscape partnership since {{year}}",
+        "profile_affiliation_card_affiliated_groups_title": "Affiliated Organizations",
+        "profile_affiliation_card_organizations_title": "Active Organizations",
+        "profile_affiliation_card_update_button": "Update Parntership Info",
+        "form_profile_partnership_status": "Is there a landscape partnership managing this landscape?",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:365]:
```suggestion
        "form_profile_partnership_status": "Is a landscape partnership managing this landscape?",
```

File: src/localization/locales/en-US.json [hunk 3]
```diff
@@ -355,7 +356,26 @@
...
+        "profile_profile_card_partnership_status_yes": "Yes",
+        "profile_profile_card_partnership_status_in_progress": "In-progress",
+         "profile_profile_card_partnership_status_no": "No",
+        "form_profile_partnership_year": "Approximate date of landscape partnership inception",
+        "form_profile_partnership_year_placeholder": "Select a year",
+        "form_profile_partnership_group": "Landscape partnership Terraso Groups",
+        "form_profile_partnership_group_placeholder": "Select Landscaoe Partnership Grroup",
+        "form_profile_affiliated_groups": "Affiliated Terraso Groups",
+        "form_profile_affiliated_groups_placeholder": "Select one or more groups",
+        "form_profile_organizations": "Main Institutions/organizations active in the landscape",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:375]:
```suggestion
        "form_profile_organizations": "Active organizations in the landscape",
```

File: src/landscape/components/LandscapeForm/AffiliationStep.js [hunk 4]
```diff
@@ -1,16 +1,162 @@
...
+      displayEmpty
+      value={field.value}
+      onChange={field.onChange}
+      labelId="TODO"
+      id="year-select"
+    >
+      <MenuItem value={''}>
+        {t('landscape.form_profile_partnership_year_placeholder')}
+      </MenuItem>
+      {_.range(1900, new Date().getFullYear() + 1)
```
Comment (paulschreiber) [inline src/landscape/components/LandscapeForm/AffiliationStep.js:104]: suggestion: put he 1900 in a constant, i.e. `PARTNERSHIP_START_YEAR`

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: src/forms/components/Form.js [hunk 5]
```diff
@@ -113,43 +114,50 @@ const Form = props => {
...
-              />
-            </Grid>
-          )
-        )}
+        {fields
+          .filter(field => {
+            if (!filterField) {
+              return true;
+            }
+            return filterField(field, formProps);
```
Comment (paulschreiber) [inline src/forms/components/Form.js:122]: would 
```php
return filterField ? filterField(field, formProps) : true;
```
or something similar make sense? Or is that too confusing?

File: src/localization/locales/en-US.json [hunk 6]
```diff
@@ -355,7 +356,26 @@
...
+        "form_profile_partnership_year": "Approximate date of landscape partnership inception",
+        "form_profile_partnership_year_placeholder": "Select a year",
+        "form_profile_partnership_group": "Landscape partnership Terraso Groups",
+        "form_profile_partnership_group_placeholder": "Select Landscaoe Partnership Grroup",
+        "form_profile_affiliated_groups": "Affiliated Terraso Groups",
+        "form_profile_affiliated_groups_placeholder": "Select one or more groups",
+        "form_profile_organizations": "Active organizations in the landscape",
+        "form_profile_organizations_placeholder": "Enter an organization name",
+        "form_affiliation_edit_title": "Update Landscape Partnership and Afflilations",
+        "form_affiliation_new_title": "Landscape Partnership and Afflilations (Optional)"
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:378]:
```suggestion
        "form_affiliation_new_title": "Landscape Partnership and Affiliations"
```

**Review by paulschreiber — APPROVED** [review]

File: src/forms/components/Form.js (hunk 5 above)
Comment (josebui) [reply src/forms/components/Form.js:122]: This makes sense too me, and personally is more readable, I will change it

**Pull request merged**